File: UVa Problems/10050 - Hartals.cpp
#include<bits/stdc++.h>
using namespace std;
bool isHoliday(int);
map<int,bool> holiday;
int main()
{
    int i,j,k,l,m,n,uni,num[100005],maxi,c,x,tc;
    bool found;
    while(cin>>tc)
    {
        for(l=1;l<=tc;l++)
        {
            set<int> ans;
            cin>>x>>n;
            for(i=0;i<n;i++)
            {
                cin>>num[i];
            }
            for(i=0;i<n;i++)
            {
                m = num[i];
                for(j=m;j<=x;j+=m)
                {
                    if(!isHoliday(j))
                        ans.insert(j);
                }
            }
            cout<<ans.size()<<endl;

        }
    }



}
bool isHoliday(int n)
{
    if(n%7==6 || n%7==0)
        return true;
    else return false;


}

File: UVa Problems/10050 - Hartals.py
def isHoliday (n) :
	if n%7==0 or n%7==6 :
		return True
	else :
		return False

num = list()
tc = int(raw_input())
for i in range(1,tc+1) :
	ans = set()
	x = int(raw_input())
	n = int(raw_input())
	for j in range(0,n) :
		m = int(raw_input())
		num.insert(j,m)
	for j in range(0,n) :
		m = num[j]
		k = m
		while k<=x :
			if not isHoliday(k) :
				ans.add(k)
			k+=m
	print (len(ans))
